### DISCUSSION TITLE
Hide articles?

### DISCUSSION THREAD

**Discussion by conradlyn** [FreshRSS/FreshRSS#5445] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

I have some feeds relating to some forum posts. Every now and then, there are posts having no actual content, but more like a notification of some actions the moderator has done to some posts. Then these "notification posts" piles up in the feeds, and flood the feed's article list.

**Describe the solution you’d like**

Is it possible to add a hide function in combination of the filter queries. For example, I would like to launch a query with keywords in the title, then hide these articles.

**Describe alternatives you’ve considered**

I now run a filter to sort all the contentless posts, then mark them as "read", but this does not remove the articles from the list.

**Comment by Alkarex** [comment, marked as answer]
The approach we can offer is to automatically mark as read the irrelevant articles of a given feed:

![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/6d2c1774-e86c-4c00-bd52-ac23b557d0e8)

**Reply by jbeltran73-2** [reply]
What about if I Only want to have unread the articles that contain some keywords? 
!keyword_I_want_to_have_in_the_article
As far as I understand in the Filter actions, if I write some keywords those articles will be marked as read, therefore I won't see them in my feed. 
Therefore I have to use the keywords in negative with above command?

Thank you

**Reply by Alkarex** [reply]
Indeed @jbeltran73-2 , negative searches would be the approach for you

**Reply by jbeltran73-2** [reply]
Thank you for your fast response

**Reply by sunjam** [reply]
Sorry, how is a negative search applied so only `Hobbit` topic titles are shown from a feed as visible and unread?

**Reply by Alkarex** [reply]
@sunjam To automatically mark as read all articles from a feed except the articles containing `Hobbit` in the title, you would use the expression `!intitle:Hobbit`
See also https://github.com/FreshRSS/FreshRSS/discussions/6087 and https://freshrss.github.io/FreshRSS/en/users/10_filter.html#with-the-search-field

**Comment by lamyergeier** [comment]
@Alkarex  [Newsboat RSS Feedreader has an option `ignore-article`](https://newsboat.org/releases/2.33/docs/newsboat.html#_killfiles), and it supports regex. For example

```
ignore-article "*" "title =~ \"Sponsor|Advertisement|[sS]horts?\""
```
 will hide all articles that have words Sponsor, Advertisement, shorts, Shorts from all feeds (indicated by `*`)

It would be nice to have regex support in removing unnecessary fields. Also we can choose the specific feed url to which this filter should be applicable.

**Reply by Alkarex** [reply]
We might have an issue somewhere already for supporting regex. It would be nice to link to it (or open a new issue if there is none)

**Reply by Alkarex** [reply]
https://github.com/FreshRSS/FreshRSS/issues/3549

**Reply by Alkarex** [reply]
Support for regex https://github.com/FreshRSS/FreshRSS/pull/6706
Tests welcome
